packages/eslint-scope/README.md

```diff
@@ -74,6 +74,91 @@ estraverse.traverse(ast, {
 });
 ```
 
+## API
+
+### ScopeManager
+
+The `ScopeManager` class is at the core of eslint-scope and is returned when you call `eslintScope.analyze()`. It manages all scopes in a given AST.
+
+#### Properties
+
+- `scopes` - An array of all scopes.
+- `globalScope` - Reference to the global scope.
+
+#### Methods
+
+- **acquire(node, inner)**
+  Acquires the appropriate scope for a given node.
+  - `node` - The AST node to acquire the scope from.
+  - `inner` - Optional boolean. When `true`, returns the innermost scope, otherwise returns the outermost scope. Default is `false`.
+  - Returns: The acquired scope or `null` if no scope is found.
+
+- **acquireAll(node)**
+  Acquires all scopes for a given node.
+  - `node` - The AST node to acquire scopes from.
+  - Returns: An array of scopes or `undefined` if none are found.
+
+- **release(node, inner)**
+  Releases the current scope and returns the upper scope.
+  - `node` - The AST node to release.
+  - `inner` - Optional boolean. When `true`, returns the innermost upper scope, otherwise returns the outermost upper scope. Default is `false`.
+  - Returns: The upper scope or `null` if no upper scope exists.
+
+- **getDeclaredVariables(node)**
+  Get variables that are declared by the node.
+  - `node` - The AST node to get declarations from.
+  - Returns: An array of variable objects declared by the node. If node doesn't declare any variables, returns an empty array.
+
+- **isGlobalReturn()**
+  Determines if global return statement should be allowed.
+  - Returns: `true` if the global return is enabled.
+
+- **isModule()**
+  Checks if the code should be handled as an ECMAScript module.
+  - Returns: `true` if the sourceType is "module".
+
+- **isImpliedStrict()**
+  Checks if implied strict mode is enabled.
+  - Returns: `true` if implied strict mode is enabled.
+
+- **isStrictModeSupported()**
+  Checks if strict mode is supported based on ECMAScript version.
+  - Returns: `true` if the ECMAScript version supports strict mode.
+
+### Scope Objects
```

Comment:
Most things are already documented in the ESLint docs: https://eslint.org/docs/latest/extend/scope-manager-interface

To avoid duplication, perhaps we could document here only what is specific to the direct use of this library, and provide a link to the ESLint docs for the rest?

Reply:
I feel like the canonical docs should be co-located with the code in some way. That way they can be kept up to date with code changes. We can also link out to the "pretty" version on the website

Note: The docs on eslint.org don't mention `implicit` at all.

Reply:
> The docs on eslint.org don't mention `implicit` at all.

Indeed, we should update those docs too. I can work on that.